### PR TITLE
Fixed wrong datamodel in Level Cluster

### PR DESCRIPTION
**Pull request by jsloth**
Added: Cluster code to LevelControlOptions Bitmap
Fixed: LevelControlOptions was defined wrong as Enum, Changed it to a Bitmap
Changed/Fixed: Level Cluster Attribute 15 (OPTIONS) datatype from BITMAP8 to LevelControlOptions (Bitmap8) type
Changed/Fixed: Level Command datatype for 'OptionsMask' and 'OptionsOverride' from BITMAP8 to LevelControlOptions

### CODE REVIEW

File: src/app/clusters/level-control/level-control.cpp [hunk 1]
```diff
@@ -408,23 +417,22 @@ static bool shouldExecuteIfOff(EndpointId endpoint, CommandId commandId, uint8_t
         // ---------- The following order is important in decision making -------
         // -----------more readable ----------
         //
-        if (optionsMask == 0xFF && optionsOverride == 0xFF)
```
Comment (jsloth) [inline src/app/clusters/level-control/level-control.cpp:411]: According to the comment [above ](https://github.com/project-chip/connectedhomeip/pull/23656/files#diff-0b846b0cf4124840f4d87631d664fed23f7529ce064d963b5021c003176516a9R404), the old check was actually wrong, as it states both fields SHALL be present or both fields SHALL be omitted. Thus changed && to ||

**Review by jmartinez-silabs — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Comment by andy31415** [comment]
@jsloth - this needs conflict resolution (zap regen)

**Comment by jsloth** [comment]
Build failure seems unrelated to my changes.

File: src/darwin/Framework/CHIP/zap-generated/MTRBaseClusters.h [hunk 2]
```diff
@@ -15982,6 +15982,11 @@ typedef NS_OPTIONS(uint32_t, MTRLevelControlFeature) {
     MTRLevelControlFeatureFrequency = 0x4,
 };
 
+typedef NS_OPTIONS(uint8_t, MTRLevelControlOptions) {
+    MTRLevelControlOptionsExecuteIfOff = 0x1,
+    MTRLevelControlOptionsCoupleColorTempToLevel = 0x2,
+};
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/zap-generated/MTRBaseClusters.h:15988]: This needs an MTR_NEWLY_AVAILABLE annotation....  If you're not comfortable with sorting out how to do that, please make sure an issue is filed, labeled "darwin", assigned to me.

Comment (jsloth) [reply src/darwin/Framework/CHIP/zap-generated/MTRBaseClusters.h:15988]: Sorry I don't know how to make that annotation (or where I should add it).
I have created an issue, but I don't have permission to add labels nor assign it. Issue created: https://github.com/project-chip/connectedhomeip/issues/23680

Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/zap-generated/MTRBaseClusters.h:15988]: Thank you, that's perfect.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
I created https://github.com/project-chip/connectedhomeip/pull/23686 to fix that timeout the Darwin job is hitting....

**Pull request merged**